### ISSUE TITLE
JSON encoding of unicode characters

### ISSUE THREAD

**Issue by anderly** [woocommerce/action-scheduler#353]
**Description:**

When using `as_schedule_single_action` and passing it args that contain international characters such as `ü`, etc., on deserialization the characters don't come back out properly encoded as the slashes (`\`) are getting dropped.

For instance, `Glückstagebuch` should get json encoded as `Gl\u00fcckstagebuch`. However, when the action callback finally runs, the slash (`\`) is missing and we just end up with `Glu00fcckstagebuch`.

**Steps to Reproduce:**

1. Make a call to ActionScheduler with international chars:

```php
as_schedule_single_action( time(), 'trigger_single_action', array( 'name' => 'Glückstagebuch' ) );
```
2. Define an action handler for the action:

```php
add_action( 'trigger_single_action', 'handle_single_action', 10, 1 );
```
3. In the action handler function, inspect the passed parameter:

```php
function handle_single_action( $name ) {
    // $name should be `Gl\u00fcckstagebuch` but instead will be `Glu00fcckstagebuch`, breaking the unicode encoding.
    var_dump( $name );
}
```

**Proposed Fix:**

In the [ActionScheduler_wpPostStore@create_post_array](https://github.com/Prospress/action-scheduler/blob/master/classes/ActionScheduler_wpPostStore.php#L35) method, `json_encode` should be called with the `JSON_UNESCAPED_UNICODE` flag like so:

```php
'post_content' => json_encode($action->get_args(), JSON_UNESCAPED_UNICODE),
```

**Comments:**

Currently, I'm having to work around it by not passing in certain params with my call to `as_schedule_single_action` and instead retrieving them in my callback. However, there are scenarios where this state may have changed since the call to `as_schedule_single_action` and I don't necessarily want the latest state, but rather the state at the time of the scheduled action call.

In my case, these are names and email addresses that I'm sending to Mailchimp. The Mailchimp API accepts unicode characters but without the slashes (`\`) or unescaped unicode, the names and email addresses end up garbled in Mailchimp with partial unicode encoding characters.

**Comment by thenbrent** [comment]
@anderly good news - this has been fixed with Action Scheduler 3.0.0 and custom tables. More details: https://github.com/woocommerce/action-scheduler/pull/371#issuecomment-544373104

**Comment by anderly** [comment]
@thenbrent awesome, thanks for the update!
